docs/articles/improve-raptor-with-rag.md

```diff
@@ -0,0 +1,561 @@
+
+![image.png](../assets/use_cases/improve-raptor-with-rag/raptor-1.png)
+
+### Key terms to look out for
+
+Before we dive in, let’s quickly review some key terms that will be useful as we explore **RAPTOR** tech. 
+
+1. **GMM Clustering**: Gaussian Mixture Models (GMM) group data into clusters based on statistical probabilities. So instead of rigidly classifying each instance into one category like K-means, GMM generates K-Gaussian distributions that consider the entire training space. This means that each point can belong to one or more distributions.
+2. **Dimensionality Reduction**: This process simplifies the data by reducing the number of variables while retaining essential features. It’s particularly important for understanding high-dimensional datasets like embeddings.
+3. **UMAP**: Uniform Manifold Approximation and Projection (UMAP) is a powerful dimensionality reduction algorithm we’ll use to shrink the size of our data point embeddings. This reduction makes it easier for clustering algorithms like GMM to cluster high-dimensional embeddings.
+4. **BIC and Elbow Method**: Both techniques help identify the optimal number of clusters in a dataset. The Bayesian Information Criterion (BIC) evaluates models based on their fit to the data while penalizing complexity. The Elbow Method plots explained variance against the number of clusters, helping to pinpoint where adding more clusters offers diminishing returns. For our purposes, we’ll leverage both methods to determine the best number of clusters.
+
+### How it actually works?
+
+Now that you’re familiar with the key terms (and if not, no worries—you’ll catch on as we go!), let’s dive into how everything actually works under the hood of RAPTOR. 
+
+- **Starting Documents (Leaves)**: RAPTOR first splits raw documents into chunks.
+- **Embedding and Clustering**: Next, we embed these chunks ([RAPTOR authors](https://arxiv.org/pdf/2401.18059) use the UMAP dimensionality reduction algorithm to minimize chunk embedding size). Then, based on their vector embeddings, we cluster our text chunks (using Gaussian Mixture Models (GMM) to ensure effective grouping of high-dimensional vector embeddings).
+- **Summarizing Clusters**: Next, we summarize the clusters of similar chunks into a node of higher-level abstractions. 
+- **Recursive Iteration**: We treat these nodes in turn as chunks, clustering them, summarizing these clusters.. iteratively, building a tree-like structure that encapsulates the relationships and hierarchical structures inherent in the raw documents we started from.
+
+![image.png](../assets/use_cases/improve-raptor-with-rag/raptor-2.png)
+
```

Comment:
Is this image yours? or taken from somewhere else?

Reply:
This is my image! I made this on my own